### DISCUSSION TITLE
Facing an issue with the `first:` pseudo-class, all the others work fine

### DISCUSSION THREAD

**Discussion by apetta** [tailwindlabs/tailwindcss#8765] [Help] [answered]
Hey guys,

I'm a bit puzzled by this one. As per title, the `first:` pseudo-class is not working for me but all the other ones are fine (`last:`, `even:`,`odd:`). 

I'm sharing a repo where I was playing around with it [here](https://github.com/apetta/next-airbnb-clone/tree/tw-pseudo-issue) _**(tw-pseudo-issue branch)**_

You can see the issue if you click the 'surprise me' button on the landing page. The component where the pseudo-classes are being used is in `/components/InfoCard.tsx` which is used in the `search.tsx` page

Would appreciate any thoughts on this!

**Comment by AM-77** [comment, marked as answer]
**The problem:** 

The [`<InfoCard>`](https://github.com/apetta/next-airbnb-clone/blob/e4566fdf82261c600ab904bc5a7ba0b49df41c78/pages/search.tsx#L52) list items are not wrapped  in a "direct" parent element, the parent element of the that list is [`<section>`](https://github.com/apetta/next-airbnb-clone/blob/e4566fdf82261c600ab904bc5a7ba0b49df41c78/pages/search.tsx#L36)  but the first child of the section is this [`<p>`](https://github.com/apetta/next-airbnb-clone/blob/e4566fdf82261c600ab904bc5a7ba0b49df41c78/pages/search.tsx#L37) which is not the first item of the InfoCard list, and it does not have the `first:` applied that's why it does not get affected.

**The solution:** 

simply wrap the [InfoCard list](https://github.com/apetta/next-airbnb-clone/blob/e4566fdf82261c600ab904bc5a7ba0b49df41c78/pages/search.tsx#L51) in a div.

**Reply by apetta** [reply]
Amazing! Thanks so much for the help, and the explanation 😄
